src/main/java/everyos.browser.webicitybrowser/everyos/browser/webicitybrowser/gui/component/OverlyingBlockComponent.java

```diff
@@ -7,7 +7,7 @@
 import everyos.engine.ribbon.core.ui.UIDirectiveWrapper;
 
 public class OverlyingBlockComponent extends BlockComponent {
-    private final Component contentView;
+    public final Component contentView;
```

Comment:
Except for the JavaDom source folder, we are leaning away from public variables.
There are still some cases were public and protected variables are used in the code, but we want to avoid adding more.
Keep this as private and, if absolutely necessary, create a getter method. However, it seems as if the code that you made this public for is commented out, so you probably don't need a getter.

Reply:
Ah yes, I realised it very late, that it was better to use layout.

Moreover, I think I can also remove the limitation of adding in only one component.

I'll look into those today

Reply:
Not fixing this issue, as a lot of classes have to be made Generic (all <Component>UI classes).
While that's not a bad thing, it's a problem for UIManager. Which only yields Component and not a possible Generic.
That can still be fixed by casting, but that's not a good idea, as we already have the casted method.